src/libraries/Microsoft.Extensions.Options.DataAnnotations/src/DataAnnotationValidateOptions.cs

```diff
@@ -35,27 +35,29 @@ public DataAnnotationValidateOptions(string name)
         public ValidateOptionsResult Validate(string name, TOptions options)
         {
             // Null name is used to configure all named options.
-            if (Name == null || name == Name)
+            if (Name != null && !string.Equals(name, Name))
             {
-                var validationResults = new List<ValidationResult>();
-                if (Validator.TryValidateObject(options,
-                    new ValidationContext(options, serviceProvider: null, items: null),
-                    validationResults,
-                    validateAllProperties: true))
-                {
-                    return ValidateOptionsResult.Success;
-                }
+                // Ignored if not validating this instance.
+                return ValidateOptionsResult.Skip;
+            }
 
-                var errors = new List<string>();
-                foreach (ValidationResult r in validationResults)
-                {
-                    errors.Add($"DataAnnotation validation failed for members: '{string.Join(",", r.MemberNames)}' with the error: '{r.ErrorMessage}'.");
-                }
-                return ValidateOptionsResult.Fail(errors);
+            var validationResults = new List<ValidationResult>();
+            if (Validator.TryValidateObject(options,
+                new ValidationContext(options, serviceProvider: null, items: null),
+                validationResults,
+                validateAllProperties: true))
+            {
+                return ValidateOptionsResult.Success;
+            }
+
+            string typeName = options.GetType().Name;
```

Comment:
could options be null here?

Reply:
No, at least not with the `Validator.TryValidateObject`. Internally it throws an `ArgumentNullException` if the options is null. I'll add the defense to the beginning of the function to ensure we're being pro-active on throwing exceptions as well to expected values.

Reply:
@maryamariyan added assertion to expected behavior. Made sure to move the assertion behind the name check to not alter any existing behaviors.